x/validator/handler_test.go

```diff
@@ -179,3 +190,409 @@ func queryValidator(setup TestSetup, owner string) (*types.Validator, error) {
+	_, err = setup.Handler(setup.Ctx, msgProposeDisableValidator)
+	require.NoError(t, err)
+
+	msgApproveDisableValidator := NewMsgApproveDisableValidator(account2.GetAddress(), valAddress)
+	_, err = setup.Handler(setup.Ctx, msgApproveDisableValidator)
+	require.Error(t, err)
+	require.ErrorIs(t, err, sdkerrors.ErrUnauthorized)
+}
+
+func TestHandler_ProposedDisableValidatorDoesNotExist(t *testing.T) {
+	setup := Setup(t)
+
+	// create Trustees
+	ba1 := authtypes.NewBaseAccount(testconstants.Address1, testconstants.PubKey1, 0, 0)
+	account1 := dclauthtypes.NewAccount(ba1,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID1)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account1)
+
+	msgApproveDisableValidator := NewMsgApproveDisableValidator(account1.GetAddress(), sdk.ValAddress(testconstants.ValidatorAddress1))
+	_, err := setup.Handler(setup.Ctx, msgApproveDisableValidator)
+	require.Error(t, err)
+	require.ErrorIs(t, err, types.ErrProposedDisableValidatorDoesNotExist)
+}
+
+func TestHandler_MessageCreatorAlreadyApprovedDisableValidator(t *testing.T) {
+	setup := Setup(t)
+
+	// create Trustees
+	ba1 := authtypes.NewBaseAccount(testconstants.Address1, testconstants.PubKey1, 0, 0)
+	account1 := dclauthtypes.NewAccount(ba1,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID1)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account1)
+
+	ba2 := authtypes.NewBaseAccount(testconstants.Address2, testconstants.PubKey2, 0, 0)
+	account2 := dclauthtypes.NewAccount(ba2,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID2)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account2)
+
+	ba3 := authtypes.NewBaseAccount(testconstants.Address3, testconstants.PubKey3, 0, 0)
+	account3 := dclauthtypes.NewAccount(ba3,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID3)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account3)
+
+	valAddress, err := sdk.ValAddressFromBech32(testconstants.ValidatorAddress1)
+	require.NoError(t, err)
+
+	// propose and approve new disablevalidator
+	msgProposeDisableValidator := NewMsgProposeDisableValidator(account1.GetAddress(), valAddress)
+	_, err = setup.Handler(setup.Ctx, msgProposeDisableValidator)
+	require.NoError(t, err)
+
+	msgApproveDisableValidator := NewMsgApproveDisableValidator(account1.GetAddress(), valAddress)
+	_, err = setup.Handler(setup.Ctx, msgApproveDisableValidator)
+	require.Error(t, err)
+	require.ErrorIs(t, err, sdkerrors.ErrUnauthorized)
+}
+
+func TestHandler_DisabledValidatorDoesNotExist(t *testing.T) {
+	setup := Setup(t)
+
+	msgEnableValidator := types.NewMsgEnableValidator(sdk.ValAddress(testconstants.Address1))
+	_, err := setup.Handler(setup.Ctx, msgEnableValidator)
+	require.Error(t, err)
+	require.ErrorIs(t, err, types.ErrDisabledValidatorDoesNotExist)
+}
+
+func TestHandler_NodeAdminCantEnableValidatorDisabledByTrustees(t *testing.T) {
```

Comment:
A Node Admin must be able to enable a validator disabled by Trustees. Either wrong test or a bug in the business logic.

Reply:
Done! ✅